### ISSUE TITLE
🐛 bug: Error: ENOENT: no such file or directory Turborepo

### ISSUE THREAD

**Issue by srizvi** [basehub-ai/basehub#74]
**npx envinfo --system --binaries** 👇🏽

```
System:
    OS: macOS 14.2.1
    CPU: (20) x64 Intel(R) Core(TM) i9-10910 CPU @ 3.60GHz
    Memory: 80.64 GB / 128.00 GB
    Shell: 5.9 - /bin/zsh
 
 Binaries:
    Node: 20.11.1 - ~/.nvm/versions/node/v20.11.1/bin/node
    npm: 10.2.4 - ~/.nvm/versions/node/v20.11.1/bin/npm
    pnpm: 8.15.4 - ~/Library/pnpm/pnpm
 ```
 
 **apps/web/package.json** 👇🏽
 
 ```
 {
  "name": "@sambi/web",
  "version": "0.1.0",
  "private": true,
  "type": "module",
  "scripts": {
    "build": "pnpm with-env next build",
    "clean": "git clean -xdf .next .turbo node_modules",
    "dev": "pnpm with-env next dev",
    "format": "prettier --check . --ignore-path ../../.gitignore",
    "lint": "dotenv -v SKIP_ENV_VALIDATION=1 next lint",
    "start": "pnpm with-env next start",
    "typecheck": "tsc --noEmit",
    "with-env": "dotenv -e ../../.env --"
  },
  "dependencies": {
    "@headlessui/react": "^1.7.13",
    "@leafac/rehype-shiki": "^2.2.1",
    "@mdx-js/loader": "^3.0.0",
    "@mdx-js/react": "^3.0.0",
    "@next/mdx": "^14.0.4",
    "@sambi/api": "workspace:^0.1.0",
    "@sambi/auth": "workspace:^0.1.0",
    "@sambi/db": "workspace:^0.1.0",
    "@sambi/ui": "workspace:^0.1.0",
    "@sambi/validators": "workspace:^0.1.0",
    "@t3-oss/env-nextjs": "^0.9.2",
    "acorn": "^8.10.0",
    "acorn-jsx": "^5.3.2",
    "basehub": "^3.0.4",
    "escape-string-regexp": "^5.0.0",
    "fast-glob": "^3.2.12",
    "framer-motion": "^11.0.5",
    "next": "^14.1.0",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "recma-import-images": "0.0.3",
    "remark-gfm": "^4.0.0",
    "remark-unwrap-images": "^4.0.0",
    "shiki": "^0.11.1",
    "unified-conditional": "0.0.2",
    "zod": "^3.22.4"
  },
  "devDependencies": {
    "@sambi/eslint-config": "workspace:^0.2.0",
    "@sambi/prettier-config": "workspace:^0.1.0",
    "@sambi/tailwind-config": "workspace:^0.1.0",
    "@sambi/tsconfig": "workspace:^0.1.0",
    "@types/mdx": "^2.0.7",
    "@types/node": "^20.11.25",
    "@types/react": "^18.2.57",
    "@types/react-dom": "^18.2.19",
    "dotenv-cli": "^7.3.0",
    "eslint": "^8.57.0",
    "jiti": "^1.21.0",
    "prettier": "^3.2.5",
    "sharp": "0.33.1",
    "tailwindcss": "^3.4.1",
    "typescript": "^5.3.3"
  },
  "eslintConfig": {
    "root": true,
    "extends": [
      "@sambi/eslint-config/base",
      "@sambi/eslint-config/nextjs",
      "@sambi/eslint-config/react"
    ]
  },
  "prettier": "@sambi/prettier-config"
}
```
 
 Apologies, didn't set up a reproduction repo for you. I'm trying to get Basehub installed on: https://github.com/sambi-dev/sambi
 inside the `apps/web` workspace. Following the docs here: https://basehub.com/docs/api-reference/basehub-sdk
 
###  Current behavior:

 - Run `pnpm basehub` or `npx basehub` from `apps/web` results in the following

 ```
 srs@SAMBIMAC web % pnpm basehub
🪄 Generating...
✔ generating the client in `/Users/srs/GitHub/sambi/apps/web/node_modules/basehub/dist/generated-client`
Error: ENOENT: no such file or directory, scandir '/Users/srs/GitHub/sambi/apps/web/node_modules/basehub/src-react-pump'
    at Object.readdirSync (node:fs:1515:26)
    at copyDirSync (/Users/srs/GitHub/sambi/node_modules/basehub/dist/bin.js:274:38)
    at writeReactPump (/Users/srs/GitHub/sambi/node_modules/basehub/dist/bin.js:268:3)
    at /Users/srs/GitHub/sambi/node_modules/basehub/dist/bin.js:353:3
    at Generator.next (<anonymous>)
    at fulfilled (/Users/srs/GitHub/sambi/node_modules/basehub/dist/bin.js:47:24) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'scandir',
  path: '/Users/srs/GitHub/sambi/apps/web/node_modules/basehub/src-react-pump'
}
```
###  Expected behavior:

- Run `pnpm basehub` or `npx basehub` from `apps/web`
- Client generated inside apps/web/node_modules and can be accessed as needed by basehub sdk
- I can get to work

### Additional information:
- Just signed up for team plan today

**Comment by srizvi** [comment]
Seem to be back in business with the canary version: `"basehub": "3.0.5-next.1",`. 

Thanks for the assistance @julianbenegas !

```
srs@SAMBIMAC web % pnpm basehub
🪄 Generating...
✔ generating the client in `/Users/srs/GitHub/sambi/apps/web/node_modules/basehub/dist/generated-client`
📦 Compiling to JavaScript...
🪄 Generated `basehub` client
srs@SAMBIMAC web % 
```

**Comment by julianbenegas** [comment]
LFG. You're welcome @srizvi. Already published a production release `"basehub": "3.0.5"` by the way :)
